### PR TITLE
[50953] Progress reporting for work package hierarchies: Change calculation and name of Work and Remaining work

### PR DESCRIPTION
**Pull request by cbliard**
Epic: https://community.openproject.org/wp/40867
Feature: https://community.openproject.org/wp/50953

This PR is for the feature and will be frequently rebased onto `dev` as the more specialized PR are merged.

Pull requests:

- [x] #14217
- [x] #14375
- [x] #14378
- [x] #14384
- [x] #14505
- [x] #14507
- [x] #14516
- and not creating PRs for the last two of this branch:
  - https://community.openproject.org/wp/50962 
  - https://community.openproject.org/wp/50963

### CODE REVIEW

**Comment by cbliard** [comment]
Removing and readding pullpreview tag to trigger a fresh pullpreview. Previous one was failing due to inability to download googleauth gem as version 1.9.0 has been yanked. It should work now that version has been updated to 1.9.1.

File: spec/support/table_helpers/column.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        raw_value.strip
+      end
+
+      def text_align
+        :ljust
+      end
+
+      def read_and_update_work_packages_data(work_packages_data)
+        work_packages_data.each do |work_package_data|
+          work_package_data => { attributes:, row: }
```
Comment (aaron-contreras) [inline spec/support/table_helpers/column.rb:101]: I had to look up what this does. Very cool that it works somewhat like a Hash destructuring.

File: spec/support/table_helpers/table.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      @work_packages_by_identifier[name]
+    end
+
+    private
+
+    def normalize_name(name)
+      symbolic_name = name.to_sym
+      return symbolic_name if @work_packages_by_identifier.has_key?(symbolic_name)
+
+      spell_checker = DidYouMean::SpellChecker.new(dictionary: @work_packages_by_identifier.keys.map(&:to_s))
```
Comment (aaron-contreras) [inline spec/support/table_helpers/table.rb:46]: I've always been curious how this is done, cool!

**Review by aaron-contreras — APPROVED** [review]
This belongs in the Hall-of-fame. Excellent 🎨

File: spec/support/table_helpers/column.rb (hunk 1 above)
Comment (cbliard) [reply spec/support/table_helpers/column.rb:101]: [pattern matching](https://docs.ruby-lang.org/en/3.2/syntax/pattern_matching_rdoc.html). I'm still not sure if I like it or not, so I try using it from times to times to get a better feeling about it.

**Pull request merged**